### DISCUSSION TITLE
How to stop changing working directory when using telescope, and possibly other plugins

### DISCUSSION THREAD

**Discussion by neonvoidx** [LazyVim/LazyVim#1645] [Q&A] [answered]
So as an example say I run the following command:
`nvim` in the directory `/home/.config`
This will open up NVIM with the working directory as expected. However if I use something like Telescopes file finder, or recent files picker etc and find a file thats in a  nested sub directory, then when i open that file, the current working directory now changes, but I would like my working directory to essentially always stay at whatever folder I open nvim at.

**Comment by dpetka2001** [comment]
If you use `<leader>fF` you shouldn't have any problems with that. Lazyvim only changes the `root_dir` according to the LSPs. So, every plugin that uses `lazyvim.util.get_root()` should behave the same for the corresponding keymaps. That's why Lazyvim provides 2 different kinds of keymaps, one for `cwd` and one for `root_dir`. If you don't like the `root_dir` behavior at all, you could disable it in `/lua/config/lazy.lua` with changing to the following
```
    {
      "LazyVim/LazyVim",
      init = function()
        require("lazyvim.util").get_root = function()
          return vim.loop.cwd()
        end
      end,
      import = "lazyvim.plugins",
    },
```

**Reply by dpetka2001** [reply]
Or you could change in your own configuration the keymaps for the plugins you want to not use `lazyvim.util.get_root()`, if you would like to keep the `get_root()` behavior around and only disable it for specific plugins.

**Reply by surmish** [reply]
I see this on doing that:

![LazyVim_get_root](https://github.com/LazyVim/LazyVim/assets/33273194/f3f3f057-5782-4a3f-922f-d9497ac4b682)

I just used `require("lazyvim.util").get_root = vim.loop.cwd` instead

**Comment by neonvoidx** [comment, marked as answer]
So it appears it was actually the `project` plugin causing this. I added the following to the `project` config which solved this issue:
```
    opts = {
      manual_mode = true,
      detection_methods = { "pattern" },
    },

```

**Comment by timjackleus** [comment]
So we already have a marked answer here but just wanted to add that this was the solution in my case for the same issue
https://github.com/LazyVim/LazyVim/discussions/1714#discussioncomment-7281079

**Reply by folke** [reply]
LazyVim's root detection never changes the cwd. It's only used for telescope, neo tree and lazygit key maps. Wherever root dir is used, there's always also a key map that works on the cwd instead.

**Reply by StevenXL** [reply]
I see that there is almost always a `cwd` and `root_dir` as you mentioned, but I find the naming to be "backwards"

Let's save I am in the `package.json` file in the following directory structure:

```plaintext
root
---level_one
------level_two
---------package.json
```

If I open `neotree using `<leader>e` (which is `root_dir`), it opens up with path set to a nested directory. If I use `<leader>E`, which is `cwd` it opens up to the `root` directory. I find the behavior backwards, like I mentioned. `cwd` should be nested and `root` should be un-nested, but this seems to do the opposite.
